core/templates/domain/question/editable-question-backend-api.service.ts

```diff
@@ -17,213 +17,273 @@
+import { QuestionDomainConstants } from 'domain/question/question-domain.constants';
+
+export interface EditableQuestionBackendResponse {
+  questionId: string;
+}
+export interface UpdateEditableQuestionBackendResponse {
+  questionDict: QuestionBackendDict;
+}
+export interface fetchQuestionResponse{
+  questionObject: Question
+}
+export interface ImageData {
+  filename: string,
+  imageBlob: Blob
+}
+@Injectable({
+  providedIn: 'root'
+})
+export class EditableQuestionBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private questionObjectFactory: QuestionObjectFactory,
+    private urlInterpolationService: UrlInterpolationService) {}
+
+  private _createQuestion(
+      skillIds: string[],
+      skillDifficulties: number[],
+      questionObject: Question,
+      imagesData: ImageData[],
+      successCallback: (value: EditableQuestionBackendResponse) => void,
+      errorCallback: (reason?: string) => void)
+    : Promise<EditableQuestionBackendResponse> {
+    return new Promise((resolve, reject) => {
+      let postData = {
+        question_dict: questionObject,
         skill_ids: skillIds,
         skill_difficulties: skillDifficulties
       };
+
       let body = new FormData();
       body.append('payload', JSON.stringify(postData));
       let filenames = imagesData.map(obj => obj.filename);
       let imageBlobs = imagesData.map(obj => obj.imageBlob);
       for (let idx in imageBlobs) {
         body.append(filenames[idx], imageBlobs[idx]);
       }
-      $http.post(QUESTION_CREATION_URL, (body), {
-        // The actual header to be added for form-data is 'multipart/form-data',
-        // But adding it manually won't work because we will miss the boundary
-        // parameter. When we keep 'Content-Type' as undefined the browser
-        // automatically fills the boundary parameter according to the form
-        // data. Refer https://stackoverflow.com/questions/37039852/. and
-        // https://stackoverflow.com/questions/34983071/.
-        // Note: This should be removed and a convetion similar to
-        // SkillCreationBackendApiService should be followed once this service
-        // is migrated to Angular 8.
-        headers: {
-          'Content-Type': undefined
-        }
-      }).then(function(response) {
-        var questionId = response.data.question_id;
-        if (successCallback) {
-          successCallback(questionId);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
-      });
-    };
-
-    var _fetchQuestion = function(questionId, successCallback, errorCallback) {
-      var questionDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_QUESTION_DATA_URL_TEMPLATE, {
+      this.http.post<EditableQuestionBackendResponse>(
+        QuestionDomainConstants.QUESTION_CREATION_URL, body)
+        .toPromise()
```

Comment:
This can be moved to the line above. In general please break a line after '('. Please do this at other places as well.